### PR TITLE
[openssl] update to 3.2.0

### PR DESCRIPTION
**Pull request by MehdiChinoune**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
jkqtplotter: baseline regression.

Downstream issues on windows: Maybe https://github.com/openssl/openssl/pull/22813/files.

freerdp: tls_free (now?) defined in openssl and in freerdp.
~~~
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libssl.a(libssl-lib-tls_common.o): in function `tls_free':
/mnt/vcpkg-ci/buildtrees/openssl/x64-linux-dbg/../src/nssl-3.2.0-de65d72060.clean/ssl/record/methods/tls_common.c:1444: multiple definition of `tls_free'; libfreerdp/libfreerdp2.a(tls.c.o):/mnt/vcpkg-ci/buildtrees/freerdp/src/2.11.2-882d1e3a25.clean/libfreerdp/crypto/tls.c:1777: first defined here
~~~

arm64-windows: openssl regression.
~~~
ecp_sm2p256-armv8.S
crypto\ec\ecp_sm2p256-armv8.S(27): fatal error C1021: invalid preprocessor command 'Load'
NMAKE : fatal error U1077: '"cl" /EP -D__ASSEMBLER__  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MDd -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP  -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 -I"crypto" -I"." -I"include" -I"providers\common\include" -I"providers\implementations\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"D:\\packages\\openssl_arm64-windows\\debug\"" -D"ENGINESDIR=\"D:\\packages\\openssl_arm64-windows\\debug\\lib\\engines-3\"" -D"MODULESDIR=\"D:\\packages\\openssl_arm64-windows\\debug\\bin\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"DEBUG" -D"_DEBUG" -D"_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE" -D"OPENSSL_SYS_WIN_CORE"   -DBSAES_ASM -DECP_NISTZ256_ASM -DECP_SM2P256_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_MONT -DOPENSSL_CPUID_OBJ -DOPENSSL_SM3_ASM -DPOLY1305_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DSM4_ASM -DVPAES_ASM -DVPSM4_ASM "crypto\ec\ecp_sm2p256-armv8.S" > crypto\ec\libcrypto-shlib-ecp_sm2p256-armv8.obj.asm' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\Hostx64\arm64\nmake.exe" /LU                 _build_libs' : return code '0x2'
~~~

**Comment by MehdiChinoune** [comment]
jkqtplotter: only on osx, I have no access to it
freerdp: issue reported https://github.com/FreeRDP/FreeRDP/issues/9590
arm64-windows: pass `no-sm2-precomp` to configure

File: ports/openssl/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "openssl",
-  "version": "3.1.4",
+  "version": "3.2.0",
   "port-version": 1,
```
Comment (Thomas1664) [inline ports/openssl/vcpkg.json:4]: Please refer to the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#versioning) on how to properly update `port-version`.

**Comment by dg0yt** [comment]
jkqtplotter: https://github.com/microsoft/vcpkg/pull/35319

File: ports/openssl/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/openssl/vcpkg.json:4]: I guess this was an oversight.
```suggestion
  "version": "3.2.0",
```

**Comment by MehdiChinoune** [comment]
arm64-windows: still failing

**Comment by dg0yt** [comment]
> arm64-windows: still failing

Set `no-asm` for arm64-windows then. asm is already off/broken for uwp and for arm32 android.

**Comment by MehdiChinoune** [comment]
A freerdp maintainer says that my patch will break ABI/API, and suggests either to update to their RC release or stick with openssl 3.1.4
https://github.com/FreeRDP/FreeRDP/issues/9590

**Comment by dg0yt** [comment]
> A freerdp maintainer says that my patch will break ABI/API, and suggests either to update to their RC release or stick with openssl 3.1.4 [FreeRDP/FreeRDP#9590](https://github.com/FreeRDP/FreeRDP/issues/9590)

We probably don't need to be concerned about the ABI breakage, given that vcpkg is for building from source. But we cannot keep the API.
Instead of the official patch, could we use public plain old macros to rename the few conflicting functions in freerdp (maybe scoped to building with OpenSSL 3.2) ?

**Pull request closed without merging**